plugins/modules/ec2_vpc_dhcp_option_info.py

```diff
@@ -181,22 +173,26 @@ def list_dhcp_options(client, module):
     if module.params.get("dhcp_options_ids"):
         params["DhcpOptionsIds"] = module.params.get("dhcp_options_ids")
 
-    try:
-        all_dhcp_options = client.describe_dhcp_options(aws_retry=True, **params)
-    except botocore.exceptions.ClientError as e:
-        module.fail_json_aws(e)
+    all_dhcp_options = describe_dhcp_options(client, **params)
+    if not all_dhcp_options and module.params.get("dhcp_options_ids"):
+        module.fail_json(msg="Requested DHCP options does not exist")
 
-    normalized_config = [
-        normalize_ec2_vpc_dhcp_config(config["DhcpConfigurations"]) for config in all_dhcp_options["DhcpOptions"]
-    ]
+    normalized_config = [normalize_ec2_vpc_dhcp_config(config["DhcpConfigurations"]) for config in all_dhcp_options]
     raw_config = [
-        camel_dict_to_snake_dict(get_dhcp_options_info(option), ignore_list=["Tags"])
-        for option in all_dhcp_options["DhcpOptions"]
+        camel_dict_to_snake_dict(
+            {
+                "DhcpOptionsId": option["DhcpOptionsId"],
+                "DhcpConfigurations": option["DhcpConfigurations"],
+                "Tags": boto3_tag_list_to_ansible_dict(option.get("Tags", [{"Value": "", "Key": "Name"}])),
```

Comment:
I'm pretty sure the default for this `get` can just be an empty list and `boto3_tag_list_to_ansible_dict` will handle it correctly, which seems a little more straightforward to me.

Reply:
@hakbailey we cannot set the default to an empty list
using `[{"Value": "", "Key": "Name"}]` as input will produce `{"Name": ""}` where the output for an empty list `[]` will produce `{}`